controller/api/serializers.py

```diff
@@ -249,9 +249,15 @@ def validate_domain(self, value):
             raise serializers.ValidationError(
                 'Adding a wildcard subdomain is currently not supported.')
         allowed = re.compile("^(?!-)[a-z0-9-]{1,63}(?<!-)$", re.IGNORECASE)
+        all_digits = True
         for label in labels:
+            if label.isalpha():
+                all_digits = False
             match = allowed.match(label)
-            if not match or '--' in label or label[-1].isdigit() or label.isdigit():
+            if not match or \
+              '--' in label or \
+              len(label) > 3 and any(char.isdigit() for char in label[1:-1]) or \
+              label.isdigit() and not all_digits:
```

Comment:
Ah, it's not going to work anyways. logic like `1.example.com` would work which is an invalid case. I'll just throw away this commit since it's not part of the original issue anyways.